contextual.go

```diff
@@ -81,10 +69,10 @@ func SetLogger(logger logr.Logger) {
 // routing log entries through klogr into klog and then into the actual Logger
 // backend.
 func SetLoggerWithOptions(logger logr.Logger, opts ...LoggerOption) {
-	globalLogger = &logger
-	globalLoggerOptions = loggerOptions{}
+	logging.logger = &logger
+	logging.loggerOptions = loggerOptions{}
 	for _, opt := range opts {
```

Comment:
Do you need to run these before "publishing" the object?  Or is there no real need because this happens one time at startup?

Reply:
`SetLoggerWithOptions` is called once at program startup and the value of these options influences how `logging.logger` is going to be used, so these option callbacks must be invoked here.